tests/test_batch.py

```diff
@@ -82,26 +99,46 @@ def test_create_batch_and_wait(self, request_mock, wait, fetch_results):
                 assert job.result == self.job_result
                 assert job.full_result == self.job_full_result
             assert len(batch.jobs) == len(batch.ordered_jobs)
-        assert request_mock.last_request.method == "GET"
+        assert mock_request.last_request.method == "GET"
 
+    @pytest.mark.usefixtures("mock_request")
     def test_get_batch(self, batch):
         batch_requested = self.sdk.get_batch(batch.id)
         assert batch_requested.id == self.batch_id
 
-    def test_batch_add_job(self, request_mock):
+    def test_batch_add_job(self, mock_request):
         batch = self.sdk.create_batch(
             serialized_sequence=self.pulser_sequence, jobs=[self.simple_job_args]
         )
         job = batch.add_job(
             runs=self.n_job_runs,
             variables=self.job_variables,
         )
-        assert request_mock.last_request.json()["batch_id"] == batch.id
+        assert mock_request.last_request.json()["batch_id"] == batch.id
         assert job.batch_id == batch.id
         assert job.runs == self.n_job_runs
         assert len(batch.ordered_jobs) == 2
 
-    def test_batch_add_job_and_wait_for_results(self, request_mock):
+    def test_batch_add_job_failure(self, request_mock, request_mock_exception):
+        request_mock.start()
+        batch = self.sdk.create_batch(
+            serialized_sequence=self.pulser_sequence, jobs=[self.simple_job_args]
+        )
+        request_mock.stop()
+        request_mock_exception.start()
```

Comment:
Looks like the purpose of this is to have a Batch object on which we can call the `add_job` method. Maybe we could simply create an instance manually (there might even be a fixture that does just that already)